platform/security/core/security-core-impl/src/main/java/ddf/security/samlp/impl/LogoutMessageImpl.java

```diff
@@ -112,16 +113,21 @@ public LogoutRequest extractSamlLogoutRequest(@NotNull String samlLogoutResponse
    *
    * @param nameIdString NameId of user to log out
    * @param issuerOrEntityId The Issuer of the LogoutRequest
+   * @param sessionIndexes
    * @return the built <code>LogoutRequest</code>
    */
   @Override
   public LogoutRequest buildLogoutRequest(
-      @NotNull String nameIdString, @NotNull String issuerOrEntityId) {
-    return buildLogoutRequest(nameIdString, issuerOrEntityId, UUID.randomUUID().toString());
+      @NotNull String nameIdString, @NotNull String issuerOrEntityId, List<String> sessionIndexes) {
+    return buildLogoutRequest(
+        nameIdString, issuerOrEntityId, UUID.randomUUID().toString(), sessionIndexes);
```

Comment:
We can use the UUID Generator service for the UUID generation instead of using `UUID.randomUUID().toString()`.  This will maintain uniformity for UUIDs within the system.
https://github.com/codice/ddf/blob/c4c2554cfa64ff7386cf4aaac6249026fce02d60/catalog/core/catalog-core-metacardgroomerplugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml#L16

Reply:
Is it thread safe?

Reply:
I believe our current implementation is thread-safe (unless i'm missing something), but there is nothing in the API that suggests any implementation needs to be. 
 https://github.com/codice/ddf/blob/c4c2554cfa64ff7386cf4aaac6249026fce02d60/platform/util/platform-util-uuidgenerator/util-uuidgenerator-impl/src/main/java/org/codice/ddf/platform/util/uuidgenerator/impl/UuidGeneratorImpl.java#L37

Reply:
I grabbed it. If I'll push it up and see if it passes integration tests.